File: src/main/java/homeWorkSix/model/Product.java
package homeWorkSix.model;

public abstract class Product {
    protected String name;
    public Product(){

    }

    @Override
    public String toString() {
        return "Product{" +
                "name='" + name + '\'' +
                '}';
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: src/main/java/homeWorkSeven/model/Product.cs
using model;

namespace model
{
        public abstract class Product
    {    
        protected string name;
        public Product(){}

        public string GetName()
        {
            return name;
        }

        public void SetName(string name) 
        {
            this.name = name;
        } 
        
    }
}